pkg/reconciler/tenancy/workspace/workspace_controller.go

```diff
@@ -119,13 +118,12 @@ type workspaceResource = committer.Resource[*tenancyv1alpha1.WorkspaceSpec, *ten
 type Controller struct {
 	queue workqueue.RateLimitingInterface
 
-	shardName                 string
-	shardExternalURL          func() string
```

Comment:
do you understand why this closure can be removed? I remember that this was about early plumbing but late computation of the value, maybe about shard object creation.

Reply:
It can be removed because instead of explicitly setting shardExternalURL in the client, we rely on the server set in the `externalLogicalClusterAdminConfig` (see `workspace_reconcile.go` below) - AFAICS this was only required before so the endpoint from `logicalClusterAdminConfig` could overridden?

Reply:
Haven't reviewed this PR, but the closure was required before because at the time the controller is created, the shard external URL isn't necessarily set/correct.

Reply:
Thanks for clarifying @ncdc - yes AFAICS it's so we can (default to genericspiserver.ExternalAddress)[https://github.com/kcp-dev/kcp/blob/main/pkg/server/config.go#L116] which presumably is not initialized at the point of creating the config.

However in this case we're no longer using the shardExternalURL at all (the server from the new external-logical-cluster-admin-kubeconfig is used instead), so I removed the argument which passes the function.

Reply:
I use it in the upcoming Sharded Syncer PR, in this commit: https://github.com/kcp-dev/kcp/pull/2675/commits/d057305bfcf7fcaa62c652dedc36ef823e2cf48a

Backgound is that the virtual workspaces (either in the embedded or standalone mode) need to know the Shard External URL, in order to use it in their logic, especially the Syncer URL (cf our discussion about exposing the shard URL to the syncer as an annotation @sttts).